openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/LightGBMModelCreator.java

```diff
@@ -383,4 +398,30 @@ private boolean schemaMatchAllFeatures(final DatasetSchema schema, final String[
 
         return isMatch;
     }
+
+    /**
+     * Number of instances per C++ train data chunk.
+     * Train data is copied from the input stream into an array of chunks.
+     * Each chunk will have this many instances. Must be set before `fit()`.
+     */
+    public long getTrainDataChunkInstancesSize() {
+        return trainDataChunkInstancesSize;
+    }
+
+    /**
+     * Number of instances per C++ train data chunk.
+     * Train data is copied from the input stream into an array of chunks.
+     * Each chunk will have this many instances. Must be set before `fit()`.
+     * <p>
+     * Although customizable, by default starts at 200k.
+     * <p>
+     * Performance overhead notes:
+     * - Too small? Performance overhead - excessive in-memory data fragmentation.
+     * - Too large? RAM overhead - in the worst case the last chunk has only 1 instance.
+     *              Each instance might have upwards of 400 features. Each costs 8 bytes.
+     *              E.g.: 100k instances of 400 features =>  320MB / chunk
+     */
+    public void setTrainDataChunkInstancesSize(final long trainDataChunkInstancesSize) {
```

Comment:
We should avoid having mutable classes as they're error prone and harder to debug: we can instead add a new constructor to this method where one can specify the chunk size